### DISCUSSION TITLE
🐞 Bug Report: Add a per-page custom css section, as opposed to the current, per element.

### DISCUSSION THREAD

**Discussion by wearenv-digital** [elementor/elementor#21496] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Im frustrated when adding custom CSS, that it is not collated in one place. I end up having to add all the css under one element, and on a page with hundreds of elements, it can get confusing as to _where_ the css is. The second frustration is that if one element has custom css applied, and then is duplicated, that css is duplicated along with the element. Having one location per page, to add the CSS would mean that the declarations are not duplicated, improving page load time. I know there is the global css section, however it does not auto update the page being worked on without reloading, like the per element custom css does.

### Describe the solution you'd like

Having one location per page, to add the CSS would mean that the declarations are not duplicated, improving page load time. I know there is the global css section, however it does not auto update the page being worked on without reloading, like the per element custom css does.

**Comment by loucasl** [comment]
There are already three places you can add custom CSS, on a widget level, on a page level and on a global level in the site settings

https://elementor.com/help/custom-css-pro/

**Reply by wearenv-digital** [reply]
I do not see where to add it on a page level. I see global level, and widget level, but not page.

**Reply by Pixlkraft** [reply]
When editing your page with Elementor, go to page settings by clicking the cog-icon in the lower left corner. Switch to the advanced tab and paste your CSS here on page level.
